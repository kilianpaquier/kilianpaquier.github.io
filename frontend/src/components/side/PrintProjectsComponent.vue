<template>
    <v-row :no-gutters="true">
        <v-col cols="12">
            <h2 class="mb-3">{{ $t("projects_title") }}</h2>
        </v-col>
        <v-col cols="12">
            <v-list class="bg-transparent">
                <v-list-item v-for="(project, index) in projects" :key="'project_' + index" :subtitle="$t(project.subtitle)" :title="project.title">
                    <template #prepend>
                        <v-avatar class="rounded-circle">
                            <v-img :src="project.icon" :eager="true" class="w-100 h-100" :cover="true" />
                        </v-avatar>
                    </template>
                </v-list-item>
            </v-list>
        </v-col>
    </v-row>
</template>

<script lang="ts">
import {defineComponent} from "vue";
import {print_projects} from "@/data/projects";

export default defineComponent({
    name: "PrintProjectsComponent",

    data: () => ({
        projects: print_projects
    })
});
</script>