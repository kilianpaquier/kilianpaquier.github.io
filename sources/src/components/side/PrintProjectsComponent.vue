<template>
  <v-container fluid>
    <!-- Projects title -->
    <h2
      class="font-weight-bold mb-3"
      v-text="$vuetify.lang.t(`$vuetify.projects-title`)"
    />

    <v-list>
      <!-- One item per lang -->
      <v-list-item v-for="(project, i) in projects" :key="i">
        <!-- Project img -->
        <v-list-item-avatar>
          <v-img
            :src="require(`@/assets/projects/list/${project.img}`)"
            eager
          />
        </v-list-item-avatar>

        <!-- Project name -->
        <v-list-item-content>
          <v-list-item-title v-text="project.title" />
          <v-list-item-subtitle
            v-text="$vuetify.lang.t(`$vuetify.${project.subtitle}`)"
          />
        </v-list-item-content>
      </v-list-item>
    </v-list>
  </v-container>
</template>

<script lang="ts">
import Vue from "vue";
import { Project } from "@/model/entities";
import Component from "vue-class-component";

@Component
export default class PrintProjectsComponent extends Vue {
  projects: Project[] = [
      {
          title: "For The Berry",
          subtitle: "projects.fortheberry.subtitle",
          img: "fortheberry.png",
      },
      {
          title: "Share Your Fit",
          subtitle: "projects.shareyourfit.subtitle",
          img: "shareyourfit.png",
      },
      {
          title: "Gestion EPI",
          subtitle: "projects.gestionepi.subtitle",
          img: "gestionepi.png",
      },
      {
          title: "T.A.GEDOFIS",
          subtitle: "projects.tagedofis.subtitle",
          img: "tagedofis.png",
      },
  ];
}
</script>
