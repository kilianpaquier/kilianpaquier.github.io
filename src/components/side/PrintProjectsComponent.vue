<template>
    <v-row :no-gutters="true">
        <v-col cols="12">
            <h2 class="mb-3">{{ $t("projects_title") }}</h2>
        </v-col>
        <v-col cols="12">
            <v-list class="bg-transparent">
                <v-list-item v-for="(project, index) in projects" :key="'project_' + index" :subtitle="$t(project.subtitle)" :title="project.title">
                    <template #prepend>
                        <v-avatar class="rounded-circle">
                            <v-img :cover="true" :eager="true" :sizes="sizes" :src="project.src" :srcset="project.srcset" class="w-100 h-100" />
                        </v-avatar>
                    </template>
                </v-list-item>
            </v-list>
        </v-col>
    </v-row>
</template>

<script lang="ts">
import {defineComponent} from "vue";
import {print_projects} from "@/data/projects";

export default defineComponent({
    name: "PrintProjectsComponent",

    setup: () => ({
        sizes: "10vw",
        projects: print_projects
    })
});
</script>