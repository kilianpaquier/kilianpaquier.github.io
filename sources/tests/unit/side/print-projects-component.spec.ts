import { createLocalVue, mount, Wrapper } from "@vue/test-utils";
import PrintProjectsComponent from "@/components/side/PrintProjectsComponent.vue";
import Vuetify from "vuetify";
import Vue from "vue";

Vue.use(Vuetify);
Vue.config.silent = true;
Vuetify.config.silent = true;

describe("PrintProjectsComponent.vue", () => {
    const localVue = createLocalVue();

    let vuetify: Vuetify;
    let wrapper: Wrapper<PrintProjectsComponent>;

    beforeEach(() => {
        vuetify = new Vuetify();

        wrapper = mount(PrintProjectsComponent, {
            localVue,
            vuetify,
        });
    });

    it("should create PrintProjectsComponent", () => {
        expect(wrapper.html()).toMatchSnapshot();
    });
});
