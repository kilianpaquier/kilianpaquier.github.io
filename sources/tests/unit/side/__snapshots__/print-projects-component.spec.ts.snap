// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrintProjectsComponent.vue should create PrintProjectsComponent 1`] = `
<div class="container container--fluid">
  <h2 class="font-weight-bold mb-3">$vuetify.projects-title</h2>
  <div role="list" class="v-list v-sheet theme--light">
    <div tabindex="-1" role="listitem" class="v-list-item theme--light">
      <div class="v-avatar v-list-item__avatar" style="height: 40px; min-width: 40px; width: 40px;">
        <div class="v-image v-responsive theme--light">
          <div class="v-responsive__content"></div>
        </div>
      </div>
      <div class="v-list-item__content">
        <div class="v-list-item__title">For The Berry</div>
        <div class="v-list-item__subtitle">$vuetify.projects.fortheberry.subtitle</div>
      </div>
    </div>
    <div tabindex="-1" role="listitem" class="v-list-item theme--light">
      <div class="v-avatar v-list-item__avatar" style="height: 40px; min-width: 40px; width: 40px;">
        <div class="v-image v-responsive theme--light">
          <div class="v-responsive__content"></div>
        </div>
      </div>
      <div class="v-list-item__content">
        <div class="v-list-item__title">Share Your Fit</div>
        <div class="v-list-item__subtitle">$vuetify.projects.shareyourfit.subtitle</div>
      </div>
    </div>
    <div tabindex="-1" role="listitem" class="v-list-item theme--light">
      <div class="v-avatar v-list-item__avatar" style="height: 40px; min-width: 40px; width: 40px;">
        <div class="v-image v-responsive theme--light">
          <div class="v-responsive__content"></div>
        </div>
      </div>
      <div class="v-list-item__content">
        <div class="v-list-item__title">Gestion EPI</div>
        <div class="v-list-item__subtitle">$vuetify.projects.gestionepi.subtitle</div>
      </div>
    </div>
    <div tabindex="-1" role="listitem" class="v-list-item theme--light">
      <div class="v-avatar v-list-item__avatar" style="height: 40px; min-width: 40px; width: 40px;">
        <div class="v-image v-responsive theme--light">
          <div class="v-responsive__content"></div>
        </div>
      </div>
      <div class="v-list-item__content">
        <div class="v-list-item__title">T.A.GEDOFIS</div>
        <div class="v-list-item__subtitle">$vuetify.projects.tagedofis.subtitle</div>
      </div>
    </div>
  </div>
</div>
`;
